### PR TITLE
pkp/pkp-lib#1503 Proposal for OJS 3.1 API specification

### PR DESCRIPTION
**Pull request by NateWr**
@kaschioudi

### CODE REVIEW

**Comment by kaschioudi** [comment]
thanks @NateWr for you work. If you don't mind instead of review the whole document at once, I will comment while implementing each entity.

**Comment by kaschioudi** [comment]
@NateWr 
Regarding Submission and SubmissionSummary definitions:
- could you please describe `urlPublished` field? Not sure what to return
- What about submission cover image? it's not present. Do we want to include it.
- Any reason why we are using `authorIds` and `galleyIds` instead of `authorSummary` and `galleySummary`.

**Comment by kaschioudi** [comment]
@NateWr : In Submission and SubmissionSummary definitions instead of `issueId` and `sectionId`, shouldn't we use `issueSummary` ?

**Comment by NateWr** [comment]
> could you please describe urlPublished field? Not sure what to return

The URL where the submission will be viewed on the frontend of the site (eg - `/article/view/1`) when published. (Editors can preview it at this URL before it's published.)

> What about submission cover image? it's not present. Do we want to include it?

Good catch! Yes, there should be a localised cover image URL as well as alt text.

> Any reason why we are using authorIds and galleyIds instead of authorSummary and galleySummary.
> In Submission and SubmissionSummary definitions instead of issueId and sectionId, shouldn't we use issueSummary ?

You're right. Let's use the summary definitions.

**Comment by NateWr** [comment]
@kaschioudi I've pushed some changes to address your questions.

File: docs/dev/swagger.yaml [hunk 1]
```diff
@@ -340,6 +396,35 @@ definitions:
...
+        type: string
+      title:
+        $ref: '#/definitions/LocaleObject'
+      description:
+        $ref: '#/definitions/LocaleObject'
+      identification:
+        type: string
+      volume:
+        type: integer
+      number:
```
Comment (kaschioudi) [inline docs/dev/swagger.yaml]: Issue::getNumber() returns string.
cf https://github.com/pkp/ojs/blob/master/classes/issue/Issue.inc.php#L84-L90

Comment (NateWr) [inline docs/dev/swagger.yaml]: Hmm, this may be one of those cases where we should coerce type. @asmecher would an Issue Number ever not be an `int`?

Comment (asmecher) [inline docs/dev/swagger.yaml]: It's not good practice, but it's possible. I suspect there are a bunch of weird variants out in the wild, e.g. `spring`, `9-10`, `3a`, etc. I'd suggest keeping it a string, odd as that seems.

Comment (NateWr) [inline docs/dev/swagger.yaml]: :+1: fcff335

**Comment by kaschioudi** [comment]
@NateWr : A galley could have more than a single file right? If so under `Galley` definition, instead of `file` we should have `files` which is going to be an array of `#/definitions/File`. what do you think?

**Comment by kaschioudi** [comment]
@NateWr Under `Author` and `AuthorSummary` even though we already have `firstName`, `middleName` and `lastName`, I suggest we add a `name` property which will return the full name as per `Identity::getFullName()`.

**Comment by NateWr** [comment]
> A galley could have more than a single file right? If so under Galley definition, instead of file we should have files which is going to be an array of #/definitions/File. what do you think?

You're right. I believe this is only the case when the galley files is a `.html` file. And the other files are attached separately. Maybe the galley needs a `file` and then a `dependentFiles` entry? I'm a little fuzzy on these details. @asmecher?

> Under Author and AuthorSummary even though we already have firstName, middleName and lastName, I suggest we add a name property which will return the full name as per Identity::getFullName()

Yes, in fact, I think we should replace `firstName`, `middleName` and `lastName` in the summary with `fullName`. The full breakdown is only needed in the complete entry.

**Comment by asmecher** [comment]
A galley has a "main" file, then potentially may dependent files (images/CSS/embedded videos/whatever). HTML and XML are the current use cases; others are likely to follow.

**Comment by NateWr** [comment]
Thanks @asmecher! :+1: 

@kaschioudi I've pushed a couple updates for Gallery and Author definitions.

**Comment by NateWr** [comment]
@kaschioudi or @asmecher any objections to me merging this as-is while we work towards implementing it for 3.1?

**Comment by kaschioudi** [comment]
no objections either.

**Pull request merged**